doc/ref_impl/cntt-ri/chapters/chapter02.md

```diff
@@ -5,15 +5,75 @@
+
+**should**: Requirements that are marked as _should_ are expected to be fulfilled by the reference architecture but it is up to each service provider to accept an implementation targeting this reference architecture that is not reflecting on any of those requirements. The same applies to _should not_.
+> RFC2119
+
+**may**: Requirements that are marked as _may_ are considered optional. The same applies to _may not_.
+
+
 <a name="2.2"></a>
-## 2.2 Heading
+## 2.2 Reference Architecture Requirement
+
+This is the reference implementation for OpenStack based CNTT RA. Please refer to [CNTT Reference Architectures:OpenStack Based:Chapter 02 - Architecture Requirement](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter02.md) for the details of RA requirement
 
+The implementation should follow all the requirement identified in the RA chapter.
 
 <a name="2.3"></a>
-## 2.3 Heading
+## 2.3 Reference Implementation Requirement
+
+The implementation **must** use open source software.
+
+The implementation **may** use software version same or different from the main branch of upstream software
+
+The implementation **must** use open source API as identified in the RA
+
+The implementation **must** provide availability and resiliency according to requirement identified in the RA
+
+The Implementation **must** be able to support VM for different profile, namely Base, Network Intensive, and Compute Intensive
+
+The Implementation **must** support automatic deployment and configuration
+
+The implementation **must** follow generic installer requirements identified in the following chapters when deploy
+
+The implementation **must** be scalable in order to fix into lab testing as well as large scale field deployment
+
+The implementation **must** provide detailed capabilities needed so as to verify test cases for Reference compliance
+
+The implementation **must** be capable of acting as 'golden NFVi' and support VNFs under certification in OVP
+
+
+<a name="2.4"></a>
+## 2.4 Typical Example for Reference Implementation
+
+The following figure explicitely shows a typical example for reference implementation based on OpenStack. Note that this is just an example, actual
+deployment can be varied in multiple aspects, including the number of VIM-ctrl, SDN/no-SDN, the number of network nodes and storage nodes and 
+leaf-spine/no leaf-spine. 
+
+
+<p align="center"><img src="../figures/chp02_typical_RI.PNG" alt="relationship" title="Document Types" width="100%"/></p>
+<p align="center"><b>Figure 1-1:</b> CNTT-RI Relationship with other communities.</p>
+
+For the purpose of reference implementation and verification, OPNFV based RI will include the following components:
+
+1 foundation node: support for building and lifecycle managing an OpenStack cloud, act as jump server and Jenkins slave in OPNFV CI
+
+3 control nodes: support for cloud control service identified in RA
```

Comment:
Minor comments:
1. Pls label Compute as Compute-1, Compute-N in the diagram for consistency. Similar comment for storage for uniformity.
2. Pls use a different colour scheme for Foundation Node for clear separation with compute node colour scheme.
3. It will be good if the diagram can show connections between Leaf and Spine
4. I quite don't agree with the title of the figure ``` CNTT-RI Relationship with other communities.```

Reply:
@msidana - Manik, please see email from me.  I can help, but need to reproduce the image file as only Fu Qiao has the master file which can be edited.

Reply:
> @msidana - Manik, please see email from me. I can help, but need to reproduce the image file as only Fu Qiao has the master file which can be edited.

Hi, Mike. I can work on the modification for the figure.